### ISSUE TITLE
Building for M1 Mac

### ISSUE THREAD

**Issue by FIrgolitsch** [allen-cell-animated/agave#153]
## Description

When building for M1 Mac I have run into several issues while following the build instructions in the readme. The first was the the nasm compiler is missing from the brew install. After installing this I could run `cmake ..` successfully. After this however the building process fails at around 75% with the following error:

```
Traceback (most recent call last):
  File "/Users/frans/code/agave/build/_deps/tensorstore-src/tools/cmake/bazel_to_cmake/bzl_library/expand_template.py", line 49, in <module>
    main()
  File "/Users/frans/code/agave/build/_deps/tensorstore-src/tools/cmake/bazel_to_cmake/bzl_library/expand_template.py", line 44, in main
    expand_template(
  File "/Users/frans/code/agave/build/_deps/tensorstore-src/tools/cmake/bazel_to_cmake/bzl_library/expand_template.py", line 32, in expand_template
    pathlib.Path(out).write_text(text, encoding='utf-8')
  File "/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/pathlib.py", line 1048, in write_text
    with self.open(mode='w', encoding=encoding, errors=errors, newline=newline) as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/pathlib.py", line 1014, in open
    return io.open(self, mode, buffering, encoding, errors, newline)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/Users/frans/code/agave/build/_deps/jpeg-build/simd/arm/neon-compat.h'
make[2]: *** [_deps/jpeg-build/simd/arm/neon-compat.h] Error 1
make[1]: *** [_deps/jpeg-build/CMakeFiles/JPEG_neon-compat_gen.dir/all] Error 2
make: *** [all] Error 2
```

I tried to use the precompiled binary as well, but that kept crashing when trying to open one of my OME-Zarr files.

## Expected Behavior

A successful build.

## Reproduction

Follow the build instructions for MacOS on an M1 Mac. Include `nasm` in the brew install command.

## Environment

Macbook Pro M1 Pro
MacOs 14.0

**Comment by toloudis** [comment]
Thank you for reporting the issue.  I'll definitely add `nasm` to the macOS build instructions in the README.   

I believe the next error is within our `tensorstore` dependency which is a pretty big part of the build.    (any errors in _deps usually are tensorstore).  I'm not sure if they claim to support mac M1 builds but will look into it.  (Sometimes removing build/_deps, and re-running cmake and rebuilding can have a good result, if you want to wait to find out.)

**Comment by FIrgolitsch** [comment]
I tried removing the _deps and rebuilding it again (`cmake` + `make) but that unfortunately didn't solve the issue. It keeps getting stuck on the `JPEG_configure_internal` step.

**Comment by toloudis** [comment]
this is now fixed by #190
